File: leetcode-1071.py
import math
class Solution:
    def gcdOfStrings(self, str1: str, str2: str) -> str:
        if str1+str2 != str2+str1:
            return ""
        gcd_base_len = math.gcd(len(str1), len(str2))
        return str1[0:gcd_base_len]


if __name__ == '__main__':
    solution = Solution()
    str1 = "ABABAB"
    str2 = "ABAB"
    print(solution.gcdOfStrings(str1,str2))

File: leetcode-1071.cpp
#include<iostream>
#include<string>
#include<numeric>

using std::string;

class Solution {
public:
    string gcdOfStrings(string str1, string str2) {
        if (str1+str2 != str2+str1) {
            return "";
        }
        int gcd_len_base = std::gcd(str1.length(), str2.length());
        return str1.substr(0, gcd_len_base);
    }
};

int main() {
    Solution solution;
    string s1{ "ABCABC" };
    string s2{ "ABC" };
    std::cout << solution.gcdOfStrings(s1,s2) << '\n';

    return 0;
}

/*
Example 1:

Input: str1 = "ABCABC", str2 = "ABC"
Output: "ABC"
Example 2:

Input: str1 = "ABABAB", str2 = "ABAB"
Output: "AB"
Example 3:

Input: str1 = "LEET", str2 = "CODE"
Output: ""


Constraints:

1 <= str1.length, str2.length <= 1000
str1 and str2 consist of English uppercase letters.
*/